lib/newrelic-roda/instrument.rb

```diff
@@ -23,14 +25,31 @@ def match_all(args)
           end
           add_method_tracer :match_all
 
+          def filter_params(params)
```

Comment:
Let's make it so that when we call `build_params` - that the result is a hash,

and at the end we can call filter params on the nice hash that `build_params` has prepared

In `build_params` - we should use `request.body` to figure out what to do - if the first non-whitespace is a `{` let's try JSON parsing for example - and if JSON parsing fails - then we'll cook something up,

Else - if nothing so far managed to parse anything - let's not send anything at all and write "unable to send params" to new relic